### PR TITLE
call analysis

### PR DESCRIPTION
**Pull request by bblay**
Automatically identifies symbol dependencies from call statements.
Does not recognise a call on a single-line if statement, of which jules has one.

### CODE REVIEW

**Review by MatthewHambley — CHANGES_REQUESTED** [review]
There is some unit testing of the analyser from data structure but this is definitely a candidate for some system testing from actual source files. This would be part of a wider system testing campaign but if it can be done piece-meal that would allow us to get some value quickly.

File: run_configs/jules/build_jules.py [hunk 1]
```diff
@@ -33,12 +35,15 @@ def jules_config(revision=None):
...
-        'conversions_mod', 'water_constants_mod', 'planet_constants_mod', 'veg_param_mod', 'flake_interface'
+        # this is on a one-line if statement, which fab doesn't currently identify
+        'imogen_update_carb',
+
+        # 'sunny', 'solpos', 'solang', 'redis', 'init_time', 'init_irrigation', 'init_urban', 'init_fire', 'init_drive',
+        # 'init_imogen', 'init_prescribed_data', 'init_vars_tmp', 'imogen_check', 'imogen_update_clim', 'control',
+        # 'imogen_update_carb', 'next_time', 'sow', 'emerge', 'develop', 'partition', 'radf_co2', 'radf_non_co2',
+        # 'adf_ch4gcm_anlg', 'drdat', 'clim_calc', 'diffcarb_land_co2', 'ocean_co2', 'diffcarb_land_ch4',
+        # 'diff_atmos_ch4', 'day_calc', 'response', 'radf_ch4', 'gcm_anlg', 'delta_temp', 'rndm', 'invert', 'vgrav',
+        # 'conversions_mod', 'water_constants_mod', 'planet_constants_mod', 'veg_param_mod', 'flake_interface'
```
[the remark this answers is not in the document]
Comment (bblay) [reply run_configs/jules/build_jules.py:46]: You mean a _valid_ reason? 😄 It was kept around for fear of losing it, but of course it'll always be in the git history. Will remove.

File: source/fab/tasks/fortran.py [hunk 2]
```diff
@@ -102,6 +102,11 @@ def run(self, hashed_file: HashedFile):
                 if obj_type == Use_Stmt:
                     self._process_use_statement(analysed_file, obj)  # raises
 
+                elif obj_type == Call_Stmt:
+                    # self._process_use_statement(analysed_file, obj)  # raises
+                    called_name = _typed_child(obj, Name)
+                    analysed_file.add_symbol_dep(called_name.string)
```
[the remark this answers is not in the document]
Comment (bblay) [reply source/fab/tasks/fortran.py:108]: For CALLed dependencies, any standard library calls _do_ get added to the set of symbol dependencies in `AnalysedFile`. However, shortly afterwards `_gen_file_deps` transforms symbol deps into file deps, and tolerates unfound symbols, as is necessary for both standard and third party libraries.

On the USE side of things, `FortranAnalyser._intrinsic_modules` defines modules to ignore, including `iso_fortran_env`.

**Review by MatthewHambley — APPROVED** [review]
This seems okay.

**Pull request merged**